### ISSUE TITLE
Feature request: Need a Function to insert external HTML Templates from WWW Source

### ISSUE THREAD

**Issue by MarkyMarkDE** [RealRaven2000/SmartTemplate4#27]
We need the ability to add external Templates from an URL something like http(s)://www.mydomain.com/mytemplate.* (such as .htm, .html or .php).
This we need for all cases means write, reply and forward.
The HTML Template should have to contain the complete html Markup with complete Meta Tags, it's not same than the current template system.

**Comment by RealRaven2000** [comment]
> We need the ability to add external Templates from an URL something like http(s)://www.mydomain.com/mytemplate.* (such as .htm, .html or .php).
> This we need for all cases means write, reply and forward.

where should the URL be entered? Could is be as an additional option in the external file templates list, so the users can mix external files and URLs? The advantage would be that I wouldn't have to generate a third UI for yet another list. 

Or should it be baked in with the normal templates? in that case maybe a variable like 

`%remoteTemplate("https://www.mydomain.com/mytemplate.php")%` which loads the template from the URL might be better.

> The HTML Template should have to contain the complete html Markup with complete Meta Tags, it's not same as the current template system.

I think you can also have a complete HTML file with the externa file templates, so it is similar from that perspective. SmartTemplate will obviously still have to download the generated HTML (and wait until the page is completely loaded(, but that from that point onward it should be pretty easy to implement.

**Comment by RealRaven2000** [comment]
> We need the ability to add external Templates from an URL something like http(s)://www.mydomain.com/mytemplate.* (such as .htm, .html or .php).
> This we need for all cases means write, reply and forward.

where should the URL be entered? Could is be as an additional option in the external file templates list, so the users can mix external files and URLs? The advantage would be that I wouldn't have to generate a third UI for yet another list. 

Or should it be baked in with the normal templates? in that case maybe a variable like 

`%remoteTemplate("https://www.mydomain.com/mytemplate.php")%` which loads the template from the URL might be better.

> The HTML Template should have to contain the complete html Markup with complete Meta Tags, it's not same as the current template system.

I think you can also have a complete HTML file with the externa file templates, so it is similar from that perspective. SmartTemplate will obviously still have to download the generated HTML (and wait until the page is completely loaded(, but that from that point onward it should be pretty easy to implement.

**Comment by MarkyMarkDE** [comment]
no, we don't need a new interface for this feature. We can add this in the templates list too (add as external template instead of internal source). But it would be great, if we can select template from list as default which adds automatically if i write/replay or forward an email with one of my identities...

**Comment by RealRaven2000** [comment]
> no, we don't need a new interface for this feature. We can add this in the templates list too (add as external template instead of internal source). But it would be great, if we can select template from list as default which adds automatically if i write/reply or forward an email with one of my identities (definied before)...

or use the `%file(url)%` command for this?

**Comment by MarkyMarkDE** [comment]
yes, maybe the better solution ;)
